### PR TITLE
feat: add goreleaser workflow

### LINKED ISSUE #296: no automatic release workflow
__Proposal:__ add a automatic release workflow in github action

__Current behavior:__ no

__Desired behavior:__ use goreleaser to build release binary automatically

__Use case:__ better release workflow

### PR DESCRIPTION
**Pull request by yihuang**
Closes #296 

## Description

Add a goreleaser github action workflow.

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: .github/workflows/goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: goreleaser
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
```
Comment (fedekunze) [inline .github/workflows/goreleaser.yml:6]:
```suggestion
     # Sequence of patterns matched against refs/tags
    tags:
      # This job will only be triggered when a tag (v.X.X.X) is pushed
      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
```

File: .github/workflows/goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    environment: release
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Set up Go
+        uses: actions/setup-go@v2
```
Comment (fedekunze) [inline .github/workflows/goreleaser.yml:16]:
```suggestion
        uses: actions/setup-go@v2.1.3
```

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      - CXX=g++
+    goos:
+      - linux
+    goarch:
+      - amd64
+    flags:
+      - -tags=cgo
+    ldflags:
+      - -s -w -X github.com/cosmos/cosmos-sdk/version.Name=ethermint -X github.com/cosmos/cosmos-sdk/version.AppName=ethermintd -X github.com/cosmos/cosmos-sdk/version.Version={{.Version}} -X github.com/cosmos/cosmos-sdk/version.Commit={{.Commit}}
+  # FIXME panicwrap 1.3.2 don't support linux-arm64
```
Comment (fedekunze) [inline .goreleaser.yml:51]:
```suggestion
  # FIXME: panicwrap 1.3.2 don't support linux-arm64
```

**Review by fedekunze — COMMENTED** [review]
this looks ok, but should we use the goreleaser action instead? 

https://github.com/goreleaser/goreleaser-action
https://github.com/cosmos/cosmos-sdk/blob/master/.github/workflows/tag.yml

**Comment by yihuang** [comment]
> this looks ok, but should we use the goreleaser action instead?
> 
> https://github.com/goreleaser/goreleaser-action
> https://goreleaser.com/ci/actions/
> https://github.com/cosmos/cosmos-sdk/blob/master/.github/workflows/tag.yml

that looks good too, when did the `RELEASE_CHANGELOG.md` get generated?

**Comment by fedekunze** [comment]
>  when did the RELEASE_CHANGELOG.md get generated?

not sure, I'd need to dig into the release process

**Comment by fedekunze** [comment]
let's use the GitHub action as it's easier to maintain

**Comment by yihuang** [comment]
> let's use the GitHub action as it's easier to maintain

done

**Review by fedekunze — APPROVED** [review]
ACK. I will wait for @khoslaventures 's review too

File: go.mod [hunk 4]
```diff
@@ -54,3 +54,6 @@ require (
 replace google.golang.org/grpc => google.golang.org/grpc v1.33.2
 
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+
+// FIXME: update after PR merged: https://github.com/bugsnag/panicwrap/pull/23
```
Comment (fedekunze) [inline go.mod:58]: can you create an issue for this?

Comment (yihuang) [reply go.mod:58]: https://github.com/tharsis/ethermint/issues/311

**Pull request merged**